src/input-select/input-range-select.tsx

```diff
@@ -0,0 +1,316 @@
+    // CONST, STATE
+    // =============================================================================
+    const [selected, setSelected] = useState<InputRangeProp<T>>({
+        from: selectedOptions?.from,
+        to: selectedOptions?.to,
+    });
+    const [showOptions, setShowOptions] = useState<InputRangeProp<boolean>>({
+        from: false,
+        to: false,
+    });
+
+    const selectorRef = useRef<HTMLButtonElement>();
+    const labelContainerRef = {
+        from: useRef<HTMLDivElement>(),
+        to: useRef<HTMLDivElement>(),
+    };
+    const [currentElementType, setCurrentElementType] = useState<
+        "start" | "end" | "none"
+    >("none");
+    // =============================================================================
+    // EFFECTS
+    // =============================================================================
+    useEffect(() => {
+        setSelected({ from: selectedOptions?.from, to: selectedOptions?.to });
+    }, [selectedOptions]);
+
+    // =============================================================================
+    // EVENT HANDLERS
+    // =============================================================================
+    const handleSelectorClick = (
+        event: React.MouseEvent,
+        rangeType?: RangeType
+    ) => {
+        event.stopPropagation();
+        event.preventDefault();
+        if (disabled || readOnly) {
+            return;
+        }
+        if (rangeType === "from") {
+            showDropDown("from");
+        } else if (rangeType === "to" && selected.from) {
+            showDropDown("to");
+        } else if (rangeType === "to" && !selected.from) {
+            showDropDown("from");
+        } else {
+            showDropDown("from");
+        }
+    };
+
+    const showDropDown = (rangeType: RangeType) => {
+        const otherRangeType = rangeType === "from" ? "to" : "from";
+        showOptions[rangeType] = true;
+        showOptions[otherRangeType] = false;
+        setShowOptions({ ...showOptions });
+        setCurrentElementType(rangeType === "from" ? "start" : "end");
+    };
+
+    const handleListItemClick = (
+        item: T,
+        extractedValue: V,
+        rangeType?: RangeType
+    ) => {
+        selected[rangeType] = item;
+        setSelected({ ...selected });
+        triggerOptionDisplayCallback(false);
+        if (selectorRef) {
+            selectorRef.current.focus();
+        }
+        if (onSelectOption) {
+            onSelectOption({ [rangeType]: item }, extractedValue);
+        }
+        if (rangeType === "from") {
+            showOptions[rangeType] = false;
+            selected.to = undefined;
+            setSelected({ ...selected });
+            setTimeout(() => {
+                showOptions.to = true;
+                setShowOptions({ ...showOptions });
+                setCurrentElementType("end");
+            }, 250); // Allow dropdown animations to complete
```

Comment:
Possible to negotiate the behaviour to not minimise when switching options?

Reply:
Done